client/my-sites/plans-features-main/index.tsx

```diff
@@ -291,9 +292,19 @@ const PlansFeaturesMain = ( {
 	const intentFromSiteMeta = usePlanIntentFromSiteMeta();
 	const planFromUpsells = usePlanFromUpsells();
 	const [ forceDefaultPlans, setForceDefaultPlans ] = useState( false );
-
 	const [ intent, setIntent ] = useState< PlansIntent | undefined >( undefined );
+
+	const { segment: intentFromSegmentationSurvey, isFetchingSegment } = useSegmentedIntent(
+		intentFromProps === 'plans-guided',
+		siteId ?? 0,
+		intent
+	);
+
 	useEffect( () => {
+		if ( intentFromProps === 'plans-guided' ) {
+			return setIntent( intentFromSegmentationSurvey );
+		}
```

Comment:
This will likely not be necessary after removing `plans-guided`. In any case, I would rethink returning early here. If this lingers, let me have a moment totake a closer look before merging pls.

Reply:
I updated it to

```
if ( flowName === 'guided' ) {
	return setIntent( intentFromSegmentationSurvey );
}
```

WDYT? I'm not sure about removing the early return. Doesn't seem safe to engage other logic that may change the intent.

Reply:
I guess you are right. I removed the `if` entirely, since the hook now just returns `undefined` when it's not activated.

Reply:
Hey. Thanks for elaborating! I already pushed a revised version before seeing your comment. Would love to hear your thoughts, I think you're more likely to agree with the current version.

Reply:
Thanks. Replied [here](https://github.com/Automattic/wp-calypso/pull/90809/files#r1620213647)